util/algorithm_identifier_test.go

```diff
@@ -131,3 +137,44 @@ func TestCheckAlgorithmIDParamNotNULL(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPublicKeyOID(t *testing.T) {
+
+	path := "../testdata/rsaSigAlgoNoNULLParam.pem"
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		panic(fmt.Sprintf(
+			"Unable to read test certificate from %q - %q "+
+				"Does a unit test have an incorrect test file name?\n",
+			path, err))
+	}
+
+	if strings.Contains(string(data), "-BEGIN CERTIFICATE-") {
+		block, _ := pem.Decode(data)
+		if block == nil {
+			panic(fmt.Sprintf(
+				"Failed to PEM decode test certificate from %q - "+
+					"Does a unit test have a buggy test cert file?\n",
+				path))
+		}
+		data = block.Bytes
+	}
+
+	cert, err := x509.ParseCertificate(data)
+	if err != nil {
+		panic(fmt.Sprintf(
+			"Failed to parse x509 test certificate from %q - "+
+				"Does a unit test have a buggy test cert file?\n",
+			path))
+	}
+
+	oid, err := GetPublicKeyOID(cert)
+
+	if err != nil {
+		t.Error("Got an error parsing public key to get the OID")
+	}
+
+	if !oid.Equal(OidRSAEncryption) {
+		t.Errorf("Expected %s,  got %s", oid.String(), OidRSAEncryption.String())
+	}
```

Comment:
I think this test case is missing coverage of the places where an `err` would be returned from `GetPublicKeyOID`. The one path that it does test (reading a `OidRsaEncrypt` oid` is already covered by the unit tests for the "e_mp_pss_in_spki" lint.

Reply:
I have created a certificate with wrong encoding in public key but then it early fails during parsing of the certificate. The errors in the functions are returned if the data of the certificate could not be parsed, however this seems to have been already checked. Would it be acceptable to have the GetPublicKeyOID function return only the OID?

Reply:
Copying form my [other comment](https://github.com/zmap/zlint/pull/376#discussion_r380738718): I think dropping this particular unit test might be the best thing.

Reply:
OK.